src/error-table.rkt

```diff
@@ -1,20 +1,365 @@
+                         (mark-erroneous! subexpr pt))]
+
+                   [(list 'cbrt.f64 arg)
+                    #:when (is-inexact? arg)
+                    (define arg-val (hash-ref exacts-hash arg))
+                    
+                    ; Under/overflow rescue:
+                    (and (or (underflow? arg)
+                             (overflow? arg))
+                         (not (= subexpr-val arg-val))
+                         (mark-erroneous! subexpr pt))]
+                   
+                   [(list '/.f64 x-ex y-ex)
+                    #:when (or (is-inexact? x-ex) (is-inexact? y-ex))
+                    (define x (hash-ref exacts-hash x-ex))
+                    (define y (hash-ref exacts-hash y-ex))
+                    (define x/y (/ x y))
+                    (define y-oflow? (overflow? y-ex))
+                    (define x-uflow? (underflow? x-ex))
+                    (define x-oflow? (overflow? x-ex))
+                    (define y-uflow? (underflow? y-ex))
+                    
+                    (cond
+                      
+                      ; Why am I doing this?
+                      [(and (= x/y 0.0) (underflow? subexpr)) #f]
+                      
+                      ; Why am I doing this?
+                      [(and (infinite? x/y) (overflow? subexpr)) #f]
+                      
+                      ; x underflows and y rescues it
+                      [(and x-uflow? (not (= subexpr-val x)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; y underflows and x rescues it
+                      [(and y-uflow? (not (nan? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; x overflows and y rescues it
+                      [(and x-oflow? (not (infinite? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; y overflows and x rescues it
+                      [(and y-oflow? (not (= (abs subexpr-val) +0.0)))
+                       (mark-erroneous! subexpr pt)]
+                      [else #f])]
+
+                   [(list '*.f64 x-ex y-ex)
+                    #:when (or (is-inexact? x-ex) (is-inexact? y-ex))
+                    (define y-oflow? (overflow? y-ex))
+                    (define x-uflow? (underflow? x-ex))
+                    (define x-oflow? (overflow? x-ex))
+                    (define y-uflow? (underflow? y-ex))
+                    (cond
+                      ; 0 * inf ~ nan, but if they rescue each other, then it should not be nan
+                      [(and x-uflow? y-oflow? (not (nan? subexpr-val)))
+                       (mark-erroneous! subexpr pt)]
+                      
+                      ; inf * 0 ~ nan, but if they rescue each other, then it should not be nan
+                      [(and x-oflow? y-uflow? (not (nan? subexpr-val)))
```

Comment:
Ideally these & division should be the same…

Reply:
They work slightly differently, for example, underflow / underflow gives us a nan but underflow * underflow gives us a zero.